From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Godl1nk <chuanzhefong@gmail.com>
Date: Tue, 17 Sep 2024 18:23:10 +0800
Subject: [PATCH] return false if pendingRunnables isEmpty to prevent
 NoSuchElementException call


diff --git a/src/main/java/net/minecraft/util/thread/BlockableEventLoop.java b/src/main/java/net/minecraft/util/thread/BlockableEventLoop.java
index f22fb84c7e7929d6c80c44b13179cf385d8a43f9..706e7aaa5a96897682b8d2dd61eca797c9ab0c41 100644
--- a/src/main/java/net/minecraft/util/thread/BlockableEventLoop.java
+++ b/src/main/java/net/minecraft/util/thread/BlockableEventLoop.java
@@ -122,6 +122,7 @@ public abstract class BlockableEventLoop<R extends Runnable> implements Profiler
         } else if (this.blockingCount == 0 && !this.shouldRun(runnable)) {
             return false;
         } else {
+            if (pendingRunnables.isEmpty()) return false;
             this.doRunTask(this.pendingRunnables.remove());
             return true;
         }
